core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java

```diff
@@ -55,20 +55,28 @@ public Blueprint convert(BlueprintRequest json) {
         } else {
             blueprint.setBlueprintText(json.getAmbariBlueprint());
         }
-        validateBlueprint(blueprint.getBlueprintText());
-        JsonNode blueprintJson;
         try {
-            blueprintJson = JsonUtil.readTree(blueprint.getBlueprintText());
-            validateBlueprintStackVersion(blueprintJson);
-            setAmbariNameAndHostGrouCount(blueprint, blueprintJson);
+            JsonNode blueprintJson = JsonUtil.readTree(blueprint.getBlueprintText());
+            if (blueprintUtils.isAmbariBlueprint(blueprintJson)) {
+                validateBlueprint(blueprintJson);
+                validateBlueprintStackVersion(blueprintJson);
+                blueprint.setAmbariName(blueprintUtils.getBlueprintName(blueprintJson));
+                blueprint.setHostGroupCount(blueprintUtils.countHostGroups(blueprintJson));
+                blueprint.setStackType(blueprintUtils.getBlueprintStackName(blueprintJson));
+                blueprint.setStackVersion(blueprintUtils.getBlueprintStackVersion(blueprintJson));
+            } else {
+                blueprint.setAmbariName(blueprintUtils.getCDHDisplayName(blueprintJson));
```

Comment:
Why we set AmbariName in case of CM?

Reply:
I don't know why it is called ambari name. It is actually the blueprint name and it is used at many places.